.github/workflows/rc.yml

```diff
@@ -233,10 +229,9 @@ jobs:
           # tries to replace /usr/local/bin/2to3 and so on and causes
           # a conflict error.
           brew update
-          for python_package in $(brew list | grep python@ | sort -r); do
+          for python_package in $(brew list | grep python@); do
             brew install --overwrite ${python_package}
           done
-          brew install --overwrite python3
```

Comment:
If we keep Intel macOS support, we should not remove these changes because `macos-15-intel` still uses `/usr/local` as Homebrew prefix: https://github.com/apache/arrow-java/actions/runs/18250044030/job/51963266710?pr=868#step:9:65

```text
To restore the stashed changes to /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core, run:
```